### DISCUSSION TITLE
Serverless Pre-Rendering Without Next.js

### DISCUSSION THREAD

**Discussion by LearnWebCode** [vercel/community#115] [Help] [answered]
Hi! I read the blog post on SPR and I haven't been this excited about a feature in a long time!

The example project uses Next.js, which is great, but for **super simple** projects I was wondering how we can configure a serverless function to handle the HTML response for a page / route? For example, I know we can create a file in the "/api" folder and that's all it takes to setup a serverless function, but is there documentation on how we can set a serverless function to be responsible for the HTML response for a given url?

Thanks!

**Comment by coetry** [comment, marked as answer]
Hi @LearnWebCode You can return an HTML string from the function and set the appropriate `content-type`

```js

module.exports = (req, res) => {
  res.setHeader('content-type', 'text/html')
  res.setHeader('cache-control', 's-maxage=3600, stale-while-revalidate')
  res.send(`
    <html>
      <body><h1>yo</h1></body>
    </html>
  `)
}
```

**Reply by LearnWebCode** [reply]
@coetry Thanks! Final question: where would that file/code live? Or how do I configure that to be served up for a given page/route? I'm familiar with having serverless functions in `/api` folder, but if I'm wanting to use it as the HTML response for a route instead of calling it client-side I'm not sure how to configure things.

**Reply by LearnWebCode** [reply]
I tried creating a `now.json` file in the root of the project and included this:

```
{
  "routes": [{ "src:": "/about", "dest": "/api/about" }]
}

```
But the deploy failed with this message "The property `src:` is not allowed in now.json when using Now 2.0 – please remove it."

**Reply by LearnWebCode** [reply]
I used this as my `now.json` and was able to get things to work:
```
{
  "rewrites": [{ "source": "/about", "destination": "/api/about" }]
}
```
Although I think I'm missing something obvious. I'm really just looking for a way to handle situations where I want server-side rendered HTML/React instead of static pre-generated content. For really simple projects I'd rather not use Next.js, so I'm curious of the best way to set this up, and if rewrites is the best way?

**Reply by cpresler** [reply]
@LearnWebCode did you ever find a solution to this? I'm currently dealing with a similar issue

**Reply by claudionunezjr** [reply]
Also looking for a final solution, but guess I'll try configuring rewrites for now.

**Reply by Pieparker** [reply]
Hey y'all, we appreciate you reaching out! Unfortunately, this topic is rather old now. If you’re exhibiting the same or a similar issue to the original poster, we'd encourage you to create your own discussion and provide as much context as you can. This helps us to ensure that discussions are relevant, fresh and free of any possibly misleading context.
